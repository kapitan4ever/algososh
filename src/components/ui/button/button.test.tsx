import { render } from '@testing-library/react'
import { Button } from './button'
import userEvent from '@testing-library/user-event'

describe('Button', () => {
	it('renders with correct text', () => {
		const { getByRole } = render(<Button text='onClick' />)
		const button = getByRole('button')
		expect(button).toMatchSnapshot()
	})
	it('renders without correct text', () => {
		const { getByRole } = render(<Button />)
		const button = getByRole('button')
		expect(button).toMatchSnapshot()
	})
	it('renders with disabled state', () => {
		const { getByRole } = render(<Button disabled />)
		const button = getByRole('button')
		expect(button).toMatchSnapshot()
	})
	it('renders with loading state', () => {
		const { getByRole } = render(<Button isLoader />)
		const button = getByRole('button')
		expect(button).toMatchSnapshot()
	})
	it('calls onClick', () => {
		const onClick = jest.fn()
		const { getByRole } = render(<Button onClick={onClick} />)
		const button = getByRole('button')
		expect(onClick).toBeCalledTimes(0)
		userEvent.click(button)
		expect(onClick).toHaveBeenCalledTimes(1)
	})
})
