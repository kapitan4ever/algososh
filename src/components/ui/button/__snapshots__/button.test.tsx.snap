// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with correct text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    onClick
  </p>
</button>
`;

exports[`Button renders with disabled state 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button renders with loading state 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button renders without correct text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
